models/models.go

```diff
@@ -31,3 +31,15 @@ type DatabaseConnection struct {
 	Type               string                 `toml:"type"`
 	ConnectionSettings map[string]interface{} `toml:"settings"`
 }
+
+type FileUploadMessage struct {
+	// Key is the unique database connection identifier
+	Key string `json:"key"`
+
+	// Path is the S3 upload key.
+	Path string `json:"path"`
+
+	// Table is the name of the Clickhouse table
```

Comment:
nit: I'd replace `Clickhouse` with `database` and `S3` (above) with `Storage` since there are multiple implementations.